cmd/jujud/agent/machine/manifolds.go

```diff
@@ -741,6 +780,12 @@ var ifController = engine.Housing{
 	},
 }.Decorate
 
+var ifRaftLeader = engine.Housing{
```

Comment:
I'm not sure how this works.

Reply:
The raft-flag worker implements a .Check() method that returns whether this node is the raft leader. The engine.Housing takes the raft-flag worker and makes the following changes to the manifold it decorates (in this case raft-clusterer):
* adds it to the manifold inputs
* wraps the Start method in a func that gets the flagWorker, calls flagWorker.Check() and bails out if it's false (otherwise continuing to call the wrapped Start).